### DISCUSSION TITLE
Pulse too short causing issue

### DISCUSSION THREAD

**Discussion by rickydou127** [arendst/Tasmota#20438] [Q&A]
Hello, I am using an ESP8266 Wemo D1 mini to receive a pulse from an Hall Sensor on D7 (GPIO13) and then send a payload via MQTT to be treated in NodeRed for a project.  As long as the pulse is long enough, everything works well.  But if I decrease the time the magnet stays in front of the hall sensor, at one point, the ESP8266 will not register the pulse.  I have tried changing the switchdebounce to 40ms without any luck.  

Is there any other options or setting I should be looking for?

thank you

Eric

**Comment by barbudor** [comment]
You may get better luck with counters as they are using interrupts.
They triggers only on the falling edge. I don't know if this is a problem for your use-case.

If you are not interrested in the count but just the event, you can use a rule on value change and publish on a dedicated topic.

**Reply by sfromis** [reply]
Verified that the Hall Effect sensor is detecting/signaling very short pulses at TTL level? If it does, using a counter is indeed a good approach. Maybe with `CounterType 1` for a pulse timer, if that works better for you.
